packages/@react-types/shared/src/dnd.d.ts

```diff
@@ -164,12 +228,16 @@ export interface DraggableCollectionProps {
   onDragStart?: (e: DraggableCollectionStartEvent) => void,
   /** Handler that is called when the dragged element is moved. */
   onDragMove?: (e: DraggableCollectionMoveEvent) => void,
+  // TODO: is dropTarget being a dropTarget or Element too broad? Would be best if it could just be one or the other, but Element will be a thing
+  // if dropping on a non-collection drop target
   /** Handler that is called when the drag operation is ended, either as a result of a drop or a cancellation. */
   onDragEnd?: (e: DraggableCollectionEndEvent) => void,
   /** A function that returns the items being dragged. */
   getItems: (keys: Set<Key>) => DragItem[],
   /** The ref of the element that will be rendered as the drag preview while dragging. */
   preview?: RefObject<DragPreviewRenderer>,
   /** Function that returns the drop operations that are allowed for the dragged items. If not provided, all drop operations are allowed. */
-  getAllowedDropOperations?: () => DropOperation[]
+  getAllowedDropOperations?: () => DropOperation[],
+  /** Handler called when items are moved out of the draggable collection or moved into a folder within the draggable collection. */
+  onRemove?: (e: DraggableCollectionRemoveEvent) => void
```

Comment:
think about name here as well, to generic:

onItemRemove? brainstorm more

Reply:
maybe `onItemsRemoved` or `onRemoveItems`